src/TestFramework/PhpUnit/CommandLine/ArgumentsAndOptionsBuilder.php

```diff
@@ -64,4 +69,36 @@ public function build(string $configPath, string $extraOptions): array
 
         return $options;
     }
+
+    /**
+     * @param TestLocation[] $tests
+     */
+    public function buildForMutant(string $configPath, string $extraOptions, array $tests): array
+    {
+        $options = $this->buildForInitialTestsRun($configPath, $extraOptions);
+
+        if (count($tests) > 0) {
+            $filterString = '/';
+            $usedTestCases = [];
+
+            foreach ($tests as $testLocation) {
+                $testCaseString = $testLocation->getMethod();
+
+                if (array_key_exists($testCaseString, $usedTestCases)) {
+                    continue;
+                }
+
+                $usedTestCases[$testCaseString] = true;
+
+                $filterString .= escapeshellcmd($testCaseString) . '|';
+            }
+
+            $filterString = rtrim($filterString, '|') . '/';
```

Comment:
Are we sure we absolutely need this computation? I mean, is there a chance we won't run this mutant ever? And what about mutants on the same line of the same file?

Reply:
didn't get you, could you please clarify what do you mean?

This code just builds `'/App\\\\Test::test_case1|App\\\\Test::test_case2/'` line, meaning to run only 2 test cases of X. Those tests cases - is 100% of all the test cases that cover particular mutated line (Mutant)